File: CONTAIN/GameViewController.m
//
//  GameViewController.m
//  CONTAIN
//
//  Created by Quinton Ashley on 4/28/15.
//  Copyright (c) 2015 Quinton Ashley. All rights reserved.
//

#import "GameViewController.h"

@implementation GameViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  SKView *skView = (SKView *)self.view;
  _scene0 = [GameScene sceneWithSize:self.view.frame.size];
  [skView presentScene:_scene0];
}

- (void)dealloc {
  [[NSNotificationCenter defaultCenter] removeObserver:self];
}

#if TARGET_OS_IPHONE

- (void)viewDidAppear:(BOOL)animated {
  [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Release any cached data, images, etc that aren't in use.
}

#endif

- (BOOL)prefersStatusBarHidden {
  return YES;
}

- (void)gameCenterViewControllerDidFinish:(nonnull GKGameCenterViewController *)gameCenterViewController {
	
}

- (BOOL)commitEditingAndReturnError:(NSError * _Nullable __autoreleasing * _Nullable)error {
	return true;
}

- (void)encodeWithCoder:(nonnull NSCoder *)aCoder {
	
}

@end

File: GameViewController.swift
//
//  GameViewController.swift
//  SwiftGame
//
//  Created by Quinton Ashley on 10/29/19.
//  Copyright © 2019 Quinton Ashley. All rights reserved.
//

import UIKit
import SpriteKit
import GameplayKit

class GameViewController: UIViewController {
  
  override func viewDidLoad() {
    super.viewDidLoad();
    
		let view = self.view as! SKView;
    let scene = GameScene(size: self.view.frame.size);
    view.presentScene(scene);
    view.ignoresSiblingOrder = true;
		
		// debugging
    view.showsFPS = true;
//    view.showsNodeCount = true;
//		view.showsPhysics = true;
//		view.showsFields = true;
  }
  
  override var shouldAutorotate: Bool {
    return true;
  }
  
  override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
		return .portrait;
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning();
    // Release any cached data, images, etc that aren't in use.
  }
  
  override var prefersStatusBarHidden: Bool {
    return true;
  }
}
